src/agent/src/main.rs

```diff
@@ -342,6 +343,13 @@ fn setup_signal_handler(logger: &Logger, sandbox: Arc<Mutex<Sandbox>>) -> Result
 
                     p.exit_code = ret;
                     let _ = unistd::close(pipe_write);
+
+                    if p.epoller.is_some() {
+                        info!(logger, "close epoller");
+                        // close the socket file to notify readStdio to close terminal specifically
+                        // in case this process's terminal has been inherited by its children.
+                        p.epoller.as_ref().unwrap().close_wfd();
```

Comment:
There is no need to do `unwrap()` here since we do `if` check either way. How about something like this:
```rust
if let Some(ref poller) = p.epoller {
    info!(logger, "close epoller");
    poller.close_wfd()
}
```